### PR TITLE
replace hubble static certs with CronJob generation

### PR DESCRIPTION
**Pull request by WeirdMachine**
According to https://github.com/kubermatic/kubermatic/pull/8548 

```release-note
Replace hubble static ca with cron job generation.
Update Cilium to v11.1.
```

### CODE REVIEW

File: addons/cni-cilium/cilium.yaml [hunk 1]
```diff
@@ -1208,4 +1235,85 @@ spec:
...
+    k8s-app: hubble-generate-certs
+spec:
+  template:
+    metadata:
+      labels:
+        k8s-app: hubble-generate-certs
+    spec:
+      containers:
+        - name: certgen
+          image: "quay.io/cilium/certgen:v0.1.5"
```
Comment (xmudrii) [inline addons/cni-cilium/cilium.yaml:1255]: This image should be parametrized like other images.

File: addons/cni-cilium/cilium.yaml [hunk 2]
```diff
@@ -1208,4 +1235,85 @@ spec:
...
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            k8s-app: hubble-generate-certs
+        spec:
+          containers:
+            - name: certgen
+              image: "quay.io/cilium/certgen:v0.1.5"
```
Comment (xmudrii) [inline addons/cni-cilium/cilium.yaml:1297]: This image should be parametrized like other images.

File: addons/cni-cilium/cilium.yaml [hunk 3]
```diff
@@ -1208,4 +1235,85 @@ spec:
...
+---
+# Source: cilium/templates/hubble/tls-cronjob/cronjob.yaml
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: hubble-generate-certs
+  namespace: kube-system
+  labels:
+    k8s-app: hubble-generate-certs
+spec:
```
Comment (xmudrii) [inline addons/cni-cilium/cilium.yaml:1285]: Should we allow this job to run on control plane nodes?

**Review by xmudrii — CHANGES_REQUESTED** [review]

File: addons/cni-cilium/cilium.yaml (hunk 3 above)
Comment (WeirdMachine) [reply addons/cni-cilium/cilium.yaml:1280]: No, they are only needed for hubble and hubble resources are not allowed to run on controle plane nodes.

**Comment by xmudrii** [comment]
Can you please extend the release note to mention that Cilium is updated to a newer version?

File: pkg/templates/images/images.go [hunk 4]
```diff
@@ -57,6 +57,7 @@ const (
 	HubbleUI
 	HubbleUIBackend
 	HubbleProxy
+	CertGen
```
Comment (xmudrii) [inline pkg/templates/images/images.go:60]: Nit: can you please rename this to CiliumCertGen, so that we know it's a Cilium-related image?

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**